src/overmap.cpp

```diff
@@ -1221,6 +1221,9 @@ overmap::overmap( int const x, int const y ) : loc( x, y )
     } catch( const std::exception &err ) {
         debugmsg( "overmap (%d,%d) failed to load: %s", loc.x, loc.y, err.what() );
     }
+
+    set_validity_from_settings();
```

Comment:
I started with it like that, but it always needs to set `allow_generation`, never any other value.
It would make sense if it was something like `x = get_from_option( "OPTION_NAME" )`, but a getter like that could be misleading.